src/js/components/PlacementConstraintsSchemaField.js

```diff
@@ -60,34 +65,42 @@ export default class PlacementConstraintsFrameworkAdapter extends Component {
   }
 
   handleBatchChange(batch) {
-    const { data } = this.props;
+    const { formData, onChange } = this.props.fieldProps;
     const newJson = batch.reduce(jsonReducer, []);
     const newData = JSON.stringify(newJson.constraints);
 
-    if (newData !== data) {
-      this.props.onChange(newData);
+    if (newData !== formData) {
+      onChange(newData);
     } else {
       this.setState({ batch });
     }
   }
 
   render() {
+    const { label, errorMessage } = this.props;
     const { batch } = this.state;
+
     const data = { constraints: batch.reduce(FormReducer) };
 
     return (
-      <BatchContainer batch={batch} onChange={this.handleBatchChange}>
-        <PlacementConstraintsPartial data={data} />
-      </BatchContainer>
+      <div>
+        <FieldLabel>
+          {label}
+        </FieldLabel>
+        <BatchContainer batch={batch} onChange={this.handleBatchChange}>
+          <PlacementConstraintsPartial data={data} />
+        </BatchContainer>
+        <FieldError>{errorMessage}</FieldError>
+      </div>
     );
   }
 }
 
-PlacementConstraintsFrameworkAdapter.defaultProps = {
+PlacementConstraintsSchemaField.defaultProps = {
   onChange() {}
 };
 
-PlacementConstraintsFrameworkAdapter.propTypes = {
+PlacementConstraintsSchemaField.propTypes = {
```

Comment:
Change the propTypes to `fieldProps` of type object. And I think make it `.required` and remove the default props.